compiler-core/src/analyse.rs

```diff
@@ -697,11 +732,19 @@ fn target_function_implementation<'a>(
     }
 }
 
-fn ensure_body_given(body: &Vec1<UntypedStatement>, location: SrcSpan) -> Result<(), Error> {
-    if body.first().is_placeholder() {
-        Err(Error::NoImplementation { location })
-    } else {
-        Ok(())
+fn is_empty_body(body: &Vec1<UntypedStatement>) -> bool {
```

Comment:
This is called a placeholder, not empty.

Reply:
Isn't placeholder _used when parsing is incomplete or when a function body is missing_?